File: C#/Trivia/Trivia/GameRunner.cs
using System;

namespace Trivia
{
    public class GameRunner
    {
        private static Player? _winner;

        public static void Main(string[] args)
        {
            IGame aGame = new GameWhichHasEnoughPlayers(new Game());

            aGame.Add("Ozzy");
            aGame.Add("Lemmy");
            aGame.Add("Tony");

            var rand = new Random();

            do
            {
                aGame.Roll(rand.Next(5) + 1);

                if (rand.Next(9) == 7)
                {
                    _winner = aGame.WrongAnswer();
                }
                else
                {
                    _winner = aGame.WasCorrectlyAnswered();
                }
            } while (_winner == default);

            Console.WriteLine(_winner + " a gagné.");
        }
    }
}

File: kotlin/src/main/kotlin/com/adaptionsoft/games/trivia/runner/GameRunner.kt
package com.adaptionsoft.games.trivia.runner

import com.adaptionsoft.games.uglytrivia.Game
import com.adaptionsoft.games.uglytrivia.Player
import com.adaptionsoft.games.uglytrivia.errors.CouldNotReplayGameError
import java.util.*

object GameRunner {
    private var notAWinner: Boolean = false

    private var lastGame: Game? = null

    fun reset() {
        lastGame = null
    }

    fun replay() {
        if (lastGame == null) {
            throw CouldNotReplayGameError()
        }
        lastGame?.let {
            runGame(
                it,
                shouldUseRandom = true,
                shouldGoInPrison = true,
                hasCorrectAnswered = true,
                shouldStopAfterNumberOfQuestions = 50
            )
        }
    }

    fun runGame(
        game: Game,
        shouldUseRandom: Boolean = true,
        shouldGoInPrison: Boolean = true,
        hasCorrectAnswered: Boolean = true, // = shouldStayInPrison,
        shouldStopAfterNumberOfQuestions: Int = 50
    ) {
        var counter = 0
        val rand = Random()


        do {
            if (shouldUseRandom) {
                game.roll(rand.nextInt(5) + 1)
                notAWinner = if (rand.nextInt(9) == 7) {
                    game.wrongAnswer()
                } else {
                    game.wasCorrectlyAnswered()
                }
            } else {
                if (hasCorrectAnswered) {
                    game.roll(7)
                } else {
                    game.roll(2)
                }
                val secondRoll = if (shouldGoInPrison) 7 else 2
                notAWinner = if (secondRoll == 7) {
                    game.wrongAnswer()
                } else {
                    game.wasCorrectlyAnswered()
                }
                counter++
            }
        } while (notAWinner && counter < shouldStopAfterNumberOfQuestions)
    }
}

fun main(args: Array<String>) {
    val game = Game()
    game.add(arrayListOf(
        Player("Chet"),
        Player("Pat"),
        Player("Sue"),
    ))
    GameRunner.runGame(game)
}
